packages/eui/src/components/datagrid/body/header/data_grid_header_row.styles.ts

```diff
@@ -25,15 +25,41 @@ export const euiDataGridHeaderStyles = (euiThemeContext: UseEuiTheme) => {
       ${logicalCSS('width', 'fit-content')}
       display: flex;
     `,
+    euiDataGridHeaderDroppable: css`
+      display: flex;
+      ${logicalCSS('width', '100%')}
+
+      *:not(:first-child)
+      .euiDataGridHeaderCellDraggable
+      .euiDataGridHeaderCell {
+        ${logicalCSS('border-left', 'none')}
+      }
+
+      &:not(:first-child)
+        .euiDataGridHeaderCellDraggable
+        .euiDataGridHeaderCell {
+        ${logicalCSS('border-left', 'none')}
+      }
```

Comment:
I'm a little confused about this CSS - it doesn't look like the first block is doing anything, just the second block? Also, we can likely reduce some specificity here

```suggestion
      &:not(:first-child) .euiDataGridHeaderCell {
        ${logicalCSS('border-left', 'none')}
      }
```